src/hashset.h

```diff
@@ -0,0 +1,205 @@
+/* Copyright (c) 2024-present, Valkey contributors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *    * Redistributions of source code must retain the above copyright notice,
+ *      this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *    * Neither the name of the copyright holder nor the names of its
+ *      contributors may be used to endorse or promote products derived from
+ *      this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef HASHSET_H
+#define HASHSET_H
+
+/* Hash table implementation.
+ *
+ * This is a cache-friendly hash table implementation. For details about the
+ * implementation and documentation of functions, se comments in hashset.c.
+ *
+ * The elements in a hashset are of a user-defined type, but an element needs to
+ * contain a key. It can represent a key-value entry, or it can be just a key,
+ * if set semantics are desired.
+ *
+ * Terminology:
+ *
+ * hashset
+ *         An instance of the data structure.
+ *
+ * key
+ *         A key used for looking up an element in the hashset.
+ *
+ * element
+ *         An element in the hashset. This may be of the same type as the key,
+ *         or a struct containing a key and other fields.
```

Comment:
```suggestion
  * element
 *         An element in the hashset. This may be of the same type as the key,
 *         or a struct containing a key and other fields.
 * key
 *         The part of the element used for looking the element up in the hashset. 
 *         May be the entire element or one of the structs within the element.
 *
```
The current wording makes it seem like the key may be a separate entity from the element. The key must be in the element though.

Reply:
Inside the hashset, we have elements that contain a key. When looking up an element using hashsetFind, you supply just a key though.

Your wording is more intuitive I suppose. I'll take it.